tests/check/check_ttu.go

```diff
@@ -0,0 +1,181 @@
+			{Object: "directs-user:both_wild_computed", Relation: "direct", User: "user:both_wild_computed"},
+			{Object: "directs-user:both_wild_computed", Relation: "direct_wild_cond", User: "user:*", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+			{Object: "directs-user:ttu_4_only_computed", Relation: "direct", User: "user:only_computed"},
+			{Object: "ttus:4_only_wild", Relation: "direct_cond_parent", User: "directs-user:ttu_4_only_wild", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+			{Object: "ttus:4_both_wild_computed", Relation: "direct_cond_parent", User: "directs-user:both_wild_computed", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+			{Object: "ttus:4_only_computed", Relation: "direct_cond_parent", User: "directs-user:ttu_4_only_computed", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+		},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "butnot_ok",
+				Tuple:       &openfgav1.TupleKey{Object: "ttus:4_only_wild", Relation: "butnot_comp_from_direct_parent", User: "user:valid"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: true,
+			},
+			{
+				Name:        "butnot_both_match",
+				Tuple:       &openfgav1.TupleKey{Object: "ttus:4_both_wild_computed", Relation: "butnot_comp_from_direct_parent", User: "user:both_wild_computed"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: false,
+			},
+			{
+				Name:        "butnot_only_computed",
+				Tuple:       &openfgav1.TupleKey{Object: "ttus:4_only_computed", Relation: "butnot_comp_from_direct_parent", User: "user:only_computed"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_object",
+				Tuple:       &openfgav1.TupleKey{Object: "ttus:invalid", Relation: "butnot_comp_from_direct_parent", User: "user:valid"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: false,
+			},
+		},
+	},
+	{
+		Name: "userset_pa_userset_ch",
+		Tuples: []*openfgav1.TupleKey{
+			{Object: "ttus:5", Relation: "userset_parent", User: "usersets-user:ttu_5"},
+			{Object: "usersets-user:ttu_5", Relation: "userset", User: "directs-user:ttu_5#direct"},
+			{Object: "directs-user:ttu_5", Relation: "direct", User: "user:valid"},
+			{Object: "directs-user:ttu_5_unconnected", Relation: "direct", User: "user:ttu_5_unconnected_directuser"},
+			{Object: "directs-user:ttu_5_unconnected_userset", Relation: "direct", User: "user:ttu_5_unconnected_userset"},
+			{Object: "usersets-user:ttu_5_unconnected_userset", Relation: "userset", User: "directs-user:ttu_5_unconnected_userset#direct"},
+		},
```

Comment:
Could we also add assertions for the case where an object has a parent (ie similar to line 151) and the parent has an associated userset, but that userset has no members so we expect `allowed=false` for any input user?

Reply:
Added both empty_userset and empty_direct